help: Improve user stream management docs

ecxtacy opened the pull request:
Fixes #26902
- Documents the procedure to subscribe / unsubscribe a user via their profile and general stream settings.
- Both methods are separated into tabs in the documentation.
- Earlier only one of the method was documented for subscribing and unsubscribing.

Link to the current [documentation](https://zulip.com/help/manage-user-stream-subscriptions)

---

**Screenshots and screen captures:**

<details>
<summary>Previous Look</summary>

<img width="600"  src="https://github.com/zulip/zulip/assets/11803841/3b806efd-66c5-4dd5-b9ff-1a5a7c9ff6db">
</details>

<details>
<summary>After updating the docs</summary>
<img width="600" src="https://github.com/zulip/zulip/assets/11803841/acc071bb-15e5-45b2-9246-30ba32a0ec42">
<img  width="600" src="https://github.com/zulip/zulip/assets/11803841/0672cb9f-457a-4652-ae8d-aeb1e8b6263b">
</details>
<!-- Describe your pull request here.-->

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->


---
<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
See https://zulip.readthedocs.io/en/latest/contributing/reviewable-prs.html for how to update your PR description:

> If you’ve updated existing documentation in your pull request, include a link to the current documentation above the screenshot of the updates. That way a reviewer can quickly access the current documentation while reviewing your changes.

ecxtacy commented:
PR description updated, I'll take care of the PR format from now on :smiley:

alya commented:
Thanks! @drrosa could you please review this one?

drrosa commented on help/manage-user-stream-subscriptions.md:
```diff
@@ -33,6 +35,17 @@ other users to a stream][configure-invites].
+
+1. Select the **Streams** tab.
+
+1. Under **Subscribe user to streams**, select the stream you would like
+   to add the user to.
```
I think we should follow a similar pattern to https://zulip.com/help/starting-a-new-topic#how-to-start-a-new-topic (**From other views** tab), and say something like:

```suggestion
1. Under **Subscribe user to streams**, select a stream from the
   dropdown list. You can start typing to filter streams.
```

drrosa reviewed (changes requested):
@ecxtacy Thanks for working on this! I left a few review comments below.

Also, your commit summary is missing a period at the end, and preferably, "Fixes #26902." should go at the end of your commit description. We follow these conventions for commit messages https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#commit-summary-part-2, so the relevant ones here are:

> Use proper capitalization and punctuation.

> Zulip’s preferred style for this is to have the final paragraph of the commit message read, e.g., Fixes #123..

ecxtacy commented:
Thank you @drrosa for your insightful review :smiley: ! I appreciate your valuable feedback and will address the suggested corrections on this. :+1:

ecxtacy commented:
@drrosa I have performed the suggested corrections, rebased and updated the branch + corrected the commit message. I will take care of the commit pattern. :smile:

drrosa reviewed (approved):
Thanks! Everything looks good to me now. Can you update the screenshots too?

ecxtacy commented:
I've updated the screenshots! 😊 Thank you for your guidance, @drrosa! It's been a nice experience working with you.

alya commented:
Looks good to me!

timabbott commented:
Merged, thanks @ecxtacy and @drrosa!
